### PR TITLE
feat(cardinal): Bash script and make targets to automate the tagging and pushing of t…

### PR DESCRIPTION
**Pull request by jerargus**
## Overview

Add the make targets: `make tag-nakama`, `make tag-sign`, and `make tag-cardinal` that will prompt the user for a new tag version for a particular project, apply the tag, and then push the tag to origin.

## Testing and Verifying

Run `make tag-sign` to run this for the sign package. Put in a custom tag like `0.0.1-test`.
Note, this will actually push a git tag to the repo. To remove the tag you created, run:

`TAG="sign/v0.0.1-test" bash -c "git tag -d \$TAG && git push --delete origin \$TAG"`

## Summary by CodeRabbit

- **New Features**
	- Introduced new automation for version tagging and release operations to streamline updates.

### CODE REVIEW

**Comment by jerargus** [comment]
@coderabbitai review

File: scripts/tag.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# 5. Pushes the new tag to origin
+
+# Ensure the script exits if any command fails
+set -e
+
+# Set the TAG_PREFIX from an environment variable, default to 'alpha' if not set
+if [ -z "$TAG_PREFIX" ]; then
+  echo "Error: TAG_PREFIX is not set." >&2
+  exit 1
+fi
```
[the remark this answers is not in the document]
Comment (jerargus) [reply scripts/tag.sh:17]: No, this script is meant to be called from make targets which will properly set the tag prefix.

File: scripts/tag.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+else
+    echo "Current tag is: $current_tag"
+    # Remove prefix and split the version number
+    version_number=${current_tag#$TAG_PREFIX}
+    IFS='.' read -ra ADDR <<< "$version_number"
+    major=${ADDR[0]}
+    minor=${ADDR[1]}
+    patch=${ADDR[2]}
+    let "patch+=1"  # Increment the patch version by default
+fi
```
[the remark this answers is not in the document]
Comment (jerargus) [reply scripts/tag.sh]: done

File: scripts/tag.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+## REMOVE THIS SUGGGESTIONS VERSION. JUST SAY DEFAULT
+# Suggest the new version
+new_version="$major.$minor.$patch"
+echo "Suggested new version: $new_version"
+
+# Ask the user for the new version components
+read -p "Enter new version [$new_version]: " input_version
+input_version=${input_version:-$new_version}  # Use the suggested version if user input is empty
+
```
[the remark this answers is not in the document]
Comment (jerargus) [reply scripts/tag.sh]: done

**Review by technicallyty — APPROVED** [review]
LGTM. tested the prompt on my end. looks cool.

**Pull request merged**